src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -427,10 +446,13 @@ def __reconstruct_semi_honest(
         Returns:
             reconstructed_value (torch.Tensor): Reconstructed value.
         """
-        shares1 = share_ptrs[0].get_shares()[0].get()
-        shares2 = share_ptrs[1].get_shares().get()
+        request = ReplicatedSharedTensor._request_and_get
```

Comment:
Maybe: We can add a check if the share_ptrs list is not empty?  else throw an error saying _"No shares were provided for reconstruction"_?